python/lsst/meas/algorithms/measureApCorr.py

```diff
@@ -235,12 +240,14 @@ def run(self, exposure, catalog):
             # Check that we have enough data points that we have at least the minimum of degrees of
             # freedom specified in the config.
             if len(subset2) - 1 < self.config.minDegreesOfFreedom:
-                raise RuntimeError("Only %d sources for calculation of aperture correction for '%s'; "
-                                   "require at least %d."
-                                   % (len(subset2), name, self.config.minDegreesOfFreedom+1))
-                apCorrMap[fluxName] = ChebyshevBoundedField(bbox, numpy.ones((1, 1), dtype=float))
-                apCorrMap[fluxSigmaName] = ChebyshevBoundedField(bbox, numpy.zeros((1, 1), dtype=float))
-                continue
+                if name in self.config.allowFailure:
+                    self.log.warn("Unable to measure aperture correction for '%s': "
+                                  "only %d sources, but require at least %d." %
+                                  (name, len(subset2), self.config.minDegreesOfFreedom+1))
+                    continue
+                raise RuntimeError("Unable to measure aperture correction for required algorithm '%s': "
+                                   "only %d sources, but require at least %d." %
+                                   (name, len(subset2), self.config.minDegreesOfFreedom+1))
 
```

Comment:
It was dead code --- it never got hit because the exception is raised first.
Moving forward, I think it's better to have no aperture correction than a fake/wrong one because in the latter case you can have a false sense of security.